base/array.jl

```diff
@@ -992,7 +993,7 @@ julia> findnext(A,3)
 """
 function findnext(A, start::Integer)
     for i = start:length(A)
-        if A[i] != 0
+        if !isequal(A[i], 0)
```

Comment:
There was a recent proposal to make this 'iszero(x)' or zero(T), since == 0 doesn't really work right for some custom types